nonce/nonce.go

```diff
@@ -81,13 +79,17 @@ func NewNonceService(scope metrics.Scope) (*NonceService, error) {
 		panic("Failure in NewGCM: " + err.Error())
 	}
 
+	if maxUsed == 0 {
```

Comment:
It seems unlikely that this value would get configured to be a negative number but YAML and humans makes for weird interactions and it seems like a small change for some extra safety:

```suggestion
	if maxUsed <= 0 {
```